sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/ModelsRepositoryAsyncClient.java

```diff
@@ -62,4 +63,64 @@
     public ModelsRepositoryServiceVersion getServiceVersion() {
         return this.serviceVersion;
     }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmi A well-formed DTDL model Id. See <a href="https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md">DTDL specs</a>. For example 'dtmi:com:example:Thermostat;1'.
+     * @return A Map containing the model definition(s) where the key is the dtmi
+     * and the value is the raw model definition string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Map<String, String>> getModels(String dtmi) {
+        return getModels(dtmi, this.defaultDependencyResolutionOption);
+    }
+
+    /**
+     * Gets a collection of model definitions.
+     *
+     * @param dtmi                 A well-formed DTDL model Id. See <a href="https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md">DTDL specs</a>. For example 'dtmi:com:example:Thermostat;1'.
+     * @param dependencyResolution A {@link ModelsDependencyResolution} value to force model resolution behavior.
+     * @return A Map containing the model definition(s) where the key is the dtmi
+     * and the value is the raw model definition string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
```

Comment:
How does the response look like with dependecy ENABLED?

Let's say the model 1 has a dependency on model 2 which has dependency on model 3

will the response be as below?
dtmiOfModel1, rawModel1
dtmiOfModel2, rawModel2
dtmiOfModel3, rawModel3

Reply:
correct. it will fetch all dependencies and include them in the map.

Reply:
I see, so there doesn't seem to be a way to determine the hierarchy I believe. Or in this example how model 1 is associated with model3.